inference-engine/tests/functional/plugin/shared/src/single_layer_tests/eltwise.cpp

```diff
@@ -7,149 +7,120 @@
-    result << "inputShapes=" << CommonTestUtils::vec2str(vec) << "_";
-    result << "targetDevice=" << dev;
-    return result.str();
+    std::tie(inputShapes, eltwiseOpType, secondaryInputType, opType, netPrecision, targetName, additional_config) = obj.param;
+    std::ostringstream results;
+
+    results << "IS=" << CommonTestUtils::vec2str(inputShapes) << "_";
+    results << "eltwiseOpType=" << eltwiseOpType << "_";
+    results << "secondaryInputType=" << secondaryInputType << "_";
+    results << "opType=" << opType << "_";
+    results << "netPRC=" << netPrecision.name() << "_";
+    results << "targetDevice=" << targetName;
+    return results.str();
 }
 
 void EltwiseLayerTest::SetUp() {
-    EltwiseOpType op;
-    ParameterInputIdx primary_input_idx;
-    InputLayerType secondary_input_type;
-    InferenceEngine::SizeVector inputShape;
+    std::vector<std::vector<size_t>> inputShapes;
     InferenceEngine::Precision netPrecision;
-    ngraph::ParameterVector parameter_inputs;
+    EltwiseParams::InputLayerType secondaryInputType;
+    EltwiseParams::OpType opType;
+    ngraph::helpers::EltwiseTypes eltwiseType;
     std::map<std::string, std::string> additional_config;
-    std::tie(op, primary_input_idx, secondary_input_type, netPrecision, inputShape, targetDevice, additional_config) = this->GetParam();
-    configuration.insert(additional_config.begin(), additional_config.end());
+    std::tie(inputShapes, eltwiseType, secondaryInputType, opType, netPrecision, targetDevice, additional_config) = this->GetParam();
     auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
 
-    std::shared_ptr<ngraph::Node> input0_node;
-    std::shared_ptr<ngraph::Node> input1_node;
-    auto primary_input = ngraph::builder::makeParams(ngPrc, { inputShape })[0];
-
-    switch (secondary_input_type) {
-    case InputLayerType::CONSTANT:
-    {
-        auto shape_total = 1;
-        for (auto dim : inputShape) {
-            shape_total *= dim;
-        }
-
-        const float min = -10;
-        const float max = 10;
-        const float range = max - min;
-        const float step = range / shape_total;
-
-        std::vector<float> const_vec(shape_total);
-        for (int i = 0; i < shape_total; i++) {
-            const_vec[i] = min + step * i;
-        }
+    std::vector<size_t> inputShape1, inputShape2;
+    if (inputShapes.size() == 1) {
+        inputShape1 = inputShape2 = inputShapes.front();
+    } else if (inputShapes.size() == 2) {
+        inputShape1 = inputShapes.front();
+        inputShape2 = inputShapes.back();
+    } else {
+        THROW_IE_EXCEPTION << "Incorrect number of input shapes";
+    }
 
-        auto const_vals = ngraph::builder::makeConstant(ngPrc, inputShape, const_vec);
-        parameter_inputs.push_back(primary_input);
+    configuration.insert(additional_config.begin(), additional_config.end());
+    auto input = ngraph::builder::makeParams(ngPrc, {inputShape1});
```

Comment:
same as with Const layers, inputs will be integer-like floats only, if possible modify input data to have real float values

Reply:
Input data generation is a common functionality for all layer tests, so I guess it's out of the scope of the PR. 

But I definitely agree with you that we have to revise our approach for data generation. Almost always data generators used in layer tests produce the float values with zero exponential part, which can mask issues on devices performing a calculation in low precision format.

Reply:
I assume you are right, is there a ticket for that? I believe it is of high importance issue from testing interface point of view